yoyodyne/evaluators.py

```diff
@@ -31,27 +36,27 @@ def __add__(self, other_eval: EvalItem) -> EvalItem:
             EvalItem.
         """
         return EvalItem(
-            self.num_correct + other_eval.num_correct,
-            self.num_predicted + other_eval.num_predicted,
+            self.per_sample_metrics + other_eval.per_sample_metrics
         )
 
-    def __radd__(self, start_val: int) -> EvalItem:
+    def __radd__(self, start_int: int) -> EvalItem:
```

Comment:
Actually I will have to think harder about this works, but iirc, basically if you want to call `sum([object1, object2])`, python first calls `__radd__` on object one to compute the left most summand, so we need to return something.

Reply:
Ok, what is actually happening here is that when we call `sum(list)`, there is a second argument for the "start" term in the summation, which defaults to 0: `sum(list, start=0)`. `__radd__` is called because `int + EvalItem` is not defined. 

So we could instead use sum like `sum(v[metric_name] for v in validation_step_outputs, EvalItem(np.array()))`---though of course this needs to allocate an empty array, which is kind of a waste.

This is hacky though and makes me think we should reduce the list of per-batch eval metrics with a different function. Maybe I can just unroll the numpy vectors of metrics in the `per_sample_metrics` and call mean there... Though I *did* like combining them in a way that returns a new EvalItem.

Reply:
I think the empty-array solution is what I'd reach for, maybe?

Reply:
Ok. Before I do that---do you think there is a more intuitive python function than `sum`/`add` for the operation we define here? Maybe `extend`, or perhaps something from `itertools` would be clearer e.g. https://docs.python.org/3/library/itertools.html#itertools.chain.from_iterable?

Reply:
Another note here:

Allocating a python List is much faster than allocating a numpy array, but numpy.mean is much faster than statistics.mean. I think it is worth using Lists in this tradeoff as I think we will call mean only once per eval epoch, whereas we will allocate Lists/arrays a lot.